src/Tribe/Commerce/PayPal/Orders/Sales.php

```diff
@@ -0,0 +1,375 @@
+<?php
+
+/**
+ * Class Tribe__Tickets__Commerce__PayPal__Orders__Sales
+ *
+ * @since TBD
+ */
+class Tribe__Tickets__Commerce__PayPal__Orders__Sales {
+
+	/**
+	 * @var \Tribe__Cache
+	 */
+	protected $cache;
+
+	/**
+	 * Tribe__Tickets__Commerce__PayPal__Orders__Sales constructor.
+	 *
+	 * @since TBD
+	 *
+	 * @param Tribe__Cache|null $cache
+	 */
+	public function __construct( Tribe__Cache $cache = null ) {
+		$this->cache = null === $cache ? new Tribe__Cache() : $cache;
+	}
+
+	/**
+	 * Returns the revenue for a single attendee.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $attendee The attendee details.
+	 *
+	 * @return int
+	 */
+	public function get_revenue_for_attendee( array $attendee ) {
+		$revenue = $this->get_unfiltered_revenue_for_attendee( $attendee );
+
+		/**
+		 * Filters the revenue for a specific attendee.
+		 *
+		 * @since TBD
+		 *
+		 * @param int   $revenue  The revenue for this attendee.
+		 * @param array $attendee The attendee details.
+		 */
+		return apply_filters( 'tribe_tickets_commerce_paypal_attendee_revenue', $revenue, $attendee );
+	}
+
+	/**
+	 * Returns the unfiltered revenue for an attendee.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $attendee
+	 *
+	 * @return int
+	 */
+	protected function get_unfiltered_revenue_for_attendee( $attendee ) {
+		$product_id = Tribe__Utils__Array::get( $attendee, 'product_id', false );
+
+		if ( ! filter_var( $product_id, FILTER_VALIDATE_INT ) ) {
```

Comment:
interesting choice -- any reason why using this method instead of more "traditional" wp style of `absint()` or `intval`? *no need to change*, just curious for my own learning

Reply:
the two functions are conversion functions, not validation functions; here I'm validating the entry; if the `product_id` was `23.89` it would pass both `absint()` and `intval()` and yield `24` or `23` depending on PHP settings. In both cases that conversion would be wrong: `product_id` **must** be an integer before any conversion happens.

Reply:
cool, thanks for elaborating!